quick-start/setup.js

```diff
@@ -0,0 +1,282 @@
+      let dockerComposeObject = yaml.load( dockerComposeFile );
+
+      dockerComposeObject.services[ 'ckeditor-cs' ].environment.LICENSE_KEY = LICENSE_KEY;
+      dockerComposeObject.services[ 'ckeditor-cs' ].environment.ENVIRONMENTS_MANAGEMENT_SECRET_KEY = ENV_SECRET;
+      dockerComposeObject.services[ 'ckeditor-cs' ].ports[ 0 ] = `${ CS_PORT }:8000`;
+      dockerComposeObject.services[ 'node-server' ].ports[ 0 ] = `${ NODE_PORT }:3000`;
+       
+      dockerComposeFile = yaml.dump( dockerComposeObject );
+      fs.writeFileSync( './docker-compose.yml', dockerComposeFile, 'utf8' );
+   } catch ( err ) {
+      throw new Error( err );
+   }
+   logger.stepInfo( CURRENT_STEP );
+}
+
+async function startDockerContainers() {
+   CURRENT_STEP = step.dockerUp;
+   CLEANUP_NEEDED  = true;
+   const dockerSpinner = logger.spinner( CURRENT_STEP );
+   dockerSpinner.start();
+
+   const spawn = require( 'child_process' ).spawn;
+   const dockerImages = spawn( "docker-compose", [ "up", "--build" ] );
+   
+   dockerImages.output = '';
+   dockerImages.stdout.on( 'data', ( data ) => {
+      dockerImages.output += data.toString();
+   } );
+
+   return new Promise( ( resolve, reject ) => {
+     
+      const serversAvailabilityCheck = setInterval( () => {
+         if ( dockerImages.output.includes( 'Server is listening on port 8000.' ) &&
+         dockerImages.output.includes( 'Node-server is listening on port 3000' ) ) {
+            clearInterval( serversAvailabilityCheck );
+            clearTimeout( timeout );
+            dockerSpinner.stop();
+            logger.stepInfo( CURRENT_STEP );
+            resolve();
+         }
+         
+         if ( dockerImages.output.includes( 'Wrong license key.' ) ) {
+            dockerSpinner.stop();
+            reject( new SetupError( error.wrongLicense ) );
+         }
+      }, 100 ); 
+
+      const timeout = setTimeout( () => {
+         dockerSpinner.stop();
+         fs.writeFileSync( './log.txt', dockerImages.output, 'utf-8' );
+         reject( new SetupError ( error.containersTimeout ) );
+      }, 10 * 60 * 1000 );
+   } );
+}
+
+async function createEnvironment() {
+   const axios = require( 'axios' );
+
+   CURRENT_STEP = step.createEnvironment;
+   const envSpinner = logger.spinner( CURRENT_STEP );
+   envSpinner.start();
+
+   const body = {
+      ip: IP_ADDR,
+      csPort: CS_PORT,
+      nodePort: NODE_PORT,
+      secret: ENV_SECRET
+   };
+
+   try {
+      await axios.post( `http://localhost:${ NODE_PORT }/init`, body );
+   }
+   catch ( err ) {
+      envSpinner.stop();
+      throw new Error( err );
+   }
+   envSpinner.stop();
+   logger.stepInfo( CURRENT_STEP );
+}
+
+function printInstructionsAfterInstallation() {
+   logger.info( `${ chalk.green( '\nInstallation complete' ) }` );
+   logger.info( `Visit ${ chalk.underline.cyan( `http://${ IP_ADDR }:${ NODE_PORT }` ) } to start collaborating` );
+}
+
+async function askForCredential( credential ) {
```

Comment:
![](https://user-images.githubusercontent.com/1795926/116688821-85925080-a9b7-11eb-9bd6-78be379c7d3f.png)

IMO descriptions for variables will be greate. I don't know what `env_secret` is, etc.